The patch for https://bugs.python.org/issue44885 was only
merged after the Bullseye version of python 3.9

--- a/tests/unittest_nodes_lineno.py
+++ b/tests/unittest_nodes_lineno.py
@@ -1,4 +1,5 @@
 import textwrap
+import sys
 
 import pytest
 
@@ -974,7 +975,7 @@
         assert (s2.lineno, s2.col_offset) == (1, 0)
         assert (s2.end_lineno, s2.end_col_offset) == (1, 29)
         assert isinstance(s2.value, nodes.Const)  # 42.1234
-        if PY39_PLUS:
+        if PY39_PLUS and (sys.version_info.minor == 9 and sys.version_info.micro > 2):
             assert (s2.value.lineno, s2.value.col_offset) == (1, 16)
             assert (s2.value.end_lineno, s2.value.end_col_offset) == (1, 23)
         else:
@@ -999,7 +1000,7 @@
         assert (s4.lineno, s4.col_offset) == (2, 0)
         assert (s4.end_lineno, s4.end_col_offset) == (2, 17)
         assert isinstance(s4.value, nodes.Name)  # 'name'
-        if PY39_PLUS:
+        if PY39_PLUS and (sys.version_info.minor == 9 and sys.version_info.micro > 2):
             assert (s4.value.lineno, s4.value.col_offset) == (2, 10)
             assert (s4.value.end_lineno, s4.value.end_col_offset) == (2, 14)
         else:
